org.osgi.test.assertj.cm/src/main/java/org/osgi/test/assertj/cm/configuration/AbstractConfigurationAssert.java

```diff
@@ -0,0 +1,232 @@
+			failWithMessage(assertjErrorMessage, actual, bundleLocation, actualBundleLocation);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual Configuration's changeCount is equal to the
+	 * given one.
+	 *
+	 * @param changeCount the given changeCount to compare the actual
+	 *            Configuration's changeCount to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's changeCount is not
+	 *             equal to the given one.
+	 */
+	public S hasChangeCount(long changeCount) {
+		// check that actual Configuration we want to make assertions on is not
+		// null.
+		isNotNull();
+
+		// overrides the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting changeCount of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>";
+
+		// check
+		long actualChangeCount = actual.getChangeCount();
+		if (actualChangeCount != changeCount) {
+			failWithMessage(assertjErrorMessage, actual, changeCount, actualChangeCount);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual Configuration's changeCount is greater then the
+	 * given one.
+	 *
+	 * @param changeCount the given changeCount to compare the actual
+	 *            Configuration's changeCount to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's changeCount is not
+	 *             greaterThen the given one.
+	 */
+	public S hasChangeCountGreater(long changeCount) {
+		// check that actual Configuration we want to make assertions on is not
+		// null.
+		isNotNull();
+
+		// overrides the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting changeCount of:\n  <%s>\nto be greater then:\n  <%s>\nbut was:\n  <%s>";
+
+		// check
+		long actualChangeCount = actual.getChangeCount();
+		if (actualChangeCount <= changeCount) {
+			failWithMessage(assertjErrorMessage, actual, changeCount, actualChangeCount);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+
+	/**
+	 * Delegates to a DictionaryAssert using the properties of the
+	 * Configuration.
+	 *
+	 * @throws AssertionError - if the actual Configuration's properties is
+	 *             null.
+	 * @return the dictionary assert
+	 */
+	public DictionaryAssert<String, Object> hasProperiesThat() {
+		isNotNull();
+		return DictionaryAssert.assertThat(actual.getProperties())
+			.as(actual + ".properties");
+	}
+
+	/**
+	 * Verifies that the actual Configuration's factoryPid is equal to the given
+	 * one.
+	 *
+	 * @param factoryPid the given factoryPid to compare the actual
+	 *            Configuration's factoryPid to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's factoryPid is not
+	 *             equal to the given one.
+	 */
+	public S hasFactoryPid(String factoryPid) {
```

Comment:
This method should be something like `hasFactoryPidEqualTo`

Reply:
renamed as requested